### PR TITLE
OCaml 5.0 release: announce and release page

### PR DESCRIPTION
**Pull request by Octachron**
This is the OCaml 5.0.0 announce slightly in advance of tomorrow release.

### CODE REVIEW

**Comment by tmattio** [comment]
Thanks @Octachron!

A few questions/suggestions:
- Are there other changes in OCaml 5 we want to include in the highlight? (the removal of deprecated functions comes to mind)
- The release notes (under `/releases/5.0.0`) contain broken links. We'd have to deploy the manual on v2.ocaml.org, including the release assets and use the absolute URLs.
- Should we add some instructions in the release notes and announcement post to setup a new opam switch with OCaml 5?

Also, we can change the homepage banner in the same PR I think, do you want to have a look @cuihtlauac maybe?

**Comment by Octachron** [comment]
Rather than focusing on just the removal of deprecated functions, I could widen the highlights to include the "OCaml 4 compatibility guarantees" from the announce in order to assert strongly that this is new major version will not split the ecosystem à la python2 versus python3. That would mean adding
```
OCaml 5 as a language is fully compatible with OCaml 4 down to the performance
characteristics of your programs. In other words, any code that works with OCaml 4 should work the same with OCaml 5. 

The currently known exceptions to this rule are:
- the removal of many long-deprecated functions and modules
- changes to the internal runtime API
- the performance of ephemerons is currently (and temporarily) strongly degraded.
```
 to the `What's new` section in the release page.

**Comment by Octachron** [comment]
It is indeed a good idea to keep the installation section in this announce since we should hope that the announce will be read by many people not familiar with opam.

I will send the manual update to `v2.ocaml.org` tomorrow morning and update the links at this point.

File: data/news/platform/ocaml-5.0.md [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+are supported.
+
+We are planning to restore the support for most previously supported architectures and operating systems, and address remaining known issues over the course of the next year.
+
+Don't let those words of caution hinder your experimentation with OCaml 5:
+OCaml 5 as a language is fully compatible with OCaml 4 down to the performance
+characteristics of your programs. In other words, any code that works with OCaml 4 should work the same with OCaml 5. 
+
+The currently known exceptions to this rule are:
+- the removal of many long-deprecated functions and modules
```
Comment (xavierleroy) [inline data/news/platform/ocaml-5.0.md:29]: "many" -> "a number of" ?   (There aren't that many.)

File: data/news/platform/ocaml-5.0.md [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+- changes to the internal runtime API
+- the performance of ephemerons is currently (and temporarily) strongly degraded.
```
Comment (xavierleroy) [inline data/news/platform/ocaml-5.0.md:31]: "strongly degraded" -> "degraded" ?

**Comment by alainfrisch** [comment]
I ask ChatGPT for a suggestion to improve the text.  Just for the fun, here is the result:



We are excited to announce the release of OCaml 5.0.0, which features long-awaited runtime support for shared memory parallelism and effect handlers. This major update marks the culmination of more than 8 years of development and required a full rewrite of the OCaml runtime environment.

As a result, OCaml 5.0.0 is considered a more experimental version of OCaml than usual. Currently, the native compiler only supports x86-64 and arm64 architectures, and Linux, BSDs, MacOS, and mingw64 on Windows are supported operating systems. We plan to restore support for most previously supported architectures and operating systems and address remaining known issues over the course of the next year.

Please note that while OCaml 5 as a language is fully compatible with OCaml 4, there are some exceptions to this rule, including the removal of many long-deprecated functions and modules, changes to the internal runtime API, and temporarily degraded performance of ephemerons. If you encounter any unexpected behaviors, please report them on the OCaml issue tracker.

During the maturing period for OCaml 5, the OCaml 4.14 branch will be maintained by regularly backporting bug fixes from the OCaml 5 branch. The next release on the OCaml 4.14 branch, OCaml 4.14.1, is expected to follow this release soon.

For a complete list of changes, please see the changelog below.

**Comment by alainfrisch** [comment]
And then, I asked "A little more enthusiastic please" -->

We are thrilled to announce the release of OCaml 5.0.0, which introduces long-awaited runtime support for shared memory parallelism and effect handlers! This major update marks the culmination of more than 8 years of development and required a full rewrite of the OCaml runtime environment.

As a result, OCaml 5.0.0 is considered a more experimental version of OCaml than usual. Currently, the native compiler only supports x86-64 and arm64 architectures, and Linux, BSDs, MacOS, and mingw64 on Windows are supported operating systems. However, we are committed to restoring support for most previously supported architectures and operating systems and addressing remaining known issues over the course of the next year.

Please note that while OCaml 5 as a language is fully compatible with OCaml 4, there are some exceptions to this rule, including the removal of many long-deprecated functions and modules, changes to the internal runtime API, and temporarily degraded performance of ephemerons. If you encounter any unexpected behaviors, please report them on the OCaml issue tracker.

During the maturing period for OCaml 5, the OCaml 4.14 branch will be maintained by regularly backporting bug fixes from the OCaml 5 branch. The next release on the OCaml 4.14 branch, OCaml 4.14.1, is expected to follow this release soon.

We encourage you to start experimenting with OCaml 5 and hope you'll be as excited about its new features as we are! For a complete list of changes, please see the changelog below.

**Comment by alainfrisch** [comment]
And ChatGPT suggests the following text to "reassure the community by asserting strongly that this is new major version will not split the ecosystem à la python2 versus python3":


We are aware that the release of a new major version of OCaml may be concerning for some members of our community, particularly with regard to potential ecosystem fragmentation. We want to assure you that this is not our intention and that we are committed to maintaining compatibility and minimizing disruption as much as possible.

OCaml 5 as a language is fully compatible with OCaml 4, meaning that any code that works with OCaml 4 should work the same with OCaml 5. The only exceptions to this rule are the removal of many long-deprecated functions and modules, changes to the internal runtime API, and temporarily degraded performance of ephemerons.

We will continue to support the OCaml 4.14 branch by regularly backporting bug fixes from the OCaml 5 branch during the maturing period for OCaml 5. The next release on the OCaml 4.14 branch, OCaml 4.14.1, is expected to follow this release soon.

We understand the importance of maintaining a stable and cohesive ecosystem and we are committed to working with the community to ensure a smooth transition to OCaml 5. If you have any questions or concerns, please don't hesitate to contact us or raise them on the OCaml issue tracker.

File: data/news/platform/ocaml-5.0.md [hunk 3]
```diff
@@ -0,0 +1,603 @@
+---
+title: Release of OCaml 5.0.0
+description: Release of OCaml 5.0.0
+date: "2022-12-16"
+tags: [ocaml, platform]
+---
+
+We have the pleasure of celebrating the birthday of Arthur C. Clarke by
```
Comment (xavierleroy) [inline data/news/platform/ocaml-5.0.md]: Among other choices there were Jane Austen and Ludwig van Beethoven...

**Review by xavierleroy — APPROVED** [review]
Lovely !  Minor comments below.

File: data/news/platform/ocaml-5.0.md (hunk 1 above)
Comment (Octachron) [reply data/news/platform/ocaml-5.0.md:29]: I count around a half-hundred removed functions in https://github.com/ocaml/ocaml/blob/trunk/Changes#L484 which does count as "many" for me.

File: data/news/platform/ocaml-5.0.md (hunk 3 above)
Comment (Octachron) [reply data/news/platform/ocaml-5.0.md]: After a plebiscite for Jane Austen, I propose to ally the classics with the moderns and go with both Jane Austen and Arthur C. Clarke.

File: data/news/platform/ocaml-5.0.md [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+
+The highlight of this new major version of OCaml is the long-awaited runtime support
+for shared memory parallelism and effect handlers.
+
+This multicore support is the culmination of more than 8 years of effort,
+and required a full rewrite of the OCaml runtime environment.
+Consequently, OCaml 5.0.0 is expected to be a more experimental version of
+OCaml than the usual OCaml releases.
+
+For instance, in this release the native compiler only supports the x86-64 and arm64 architectures. In terms of operating systems, Linux, BSDs, MacOS and mingw64 on Windows
```
Comment (avsm) [inline data/news/platform/ocaml-5.0.md]:
```suggestion
In this release, the native compiler only supports the x86-64 and arm64 architectures. In terms of operating systems, Linux, the BSDs, macOS and mingw64 on Windows
```

File: data/news/platform/ocaml-5.0.md [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+description: Release of OCaml 5.0.0
+date: "2022-12-16"
+tags: [ocaml, platform]
+---
+
+We have the pleasure of celebrating the birthday of Arthur C. Clarke by
+announcing the release of OCaml version 5.0.0.
+
+The highlight of this new major version of OCaml is the long-awaited runtime support
+for shared memory parallelism and effect handlers.
```
Comment (avsm) [inline data/news/platform/ocaml-5.0.md]: link to the relevant manual pages here?

File: data/news/platform/ocaml-5.0.md [hunk 6]
```diff
@@ -0,0 +1,603 @@
...
+OCaml 5 as a language is fully compatible with OCaml 4 down to the performance
+characteristics of your programs. In other words, any code that works with OCaml 4 should work the same with OCaml 5. 
+
+The currently known exceptions to this rule are:
+- the removal of many long-deprecated functions and modules
+- changes to the internal runtime API
+- the performance of ephemerons is currently (and temporarily) strongly degraded.
+
+Please report any unexpected behaviours on the [OCaml issue tracker]
+(https://github.com/ocaml/ocaml/issues).
```
Comment (avsm) [inline data/news/platform/ocaml-5.0.md]: "and post any questions or comments you might have on our [discussion forums](https://discuss.ocaml.org)"

File: data/news/platform/ocaml-5.0.md [hunk 7]
```diff
@@ -0,0 +1,603 @@
...
+
+
+During the maturing period for OCaml 5, the OCaml 4.14 branch will be maintained
+by backporting bug fixes from the OCaml 5 branch at regular intervals.
+The next release on the OCaml 4.14 branch, OCaml 4.14.1, should soon follow this release.
+
+The full list of changes can be found in the changelog below.
+
+Those releases are available as opam switches, and as a source download on
+[Github](https://github.com/ocaml/ocaml/archive/5.0.0.tar.gz) and on the [Inria archive]
```
Comment (avsm) [inline data/news/platform/ocaml-5.0.md]:
```suggestion
[GitHub](https://github.com/ocaml/ocaml/archive/5.0.0.tar.gz) and on the [Inria archive]
```

**Review by avsm — APPROVED** [review]

**Comment by Octachron** [comment]
The release is available on opam, the announce can be merged now.

**Pull request merged**